text/0020-distingush_manual_vs_scripted_transforms.md

```diff
@@ -0,0 +1,312 @@
+
+> The role of dataset.yml was a bit confusing to me because it seems to have a few responsibilities some of which are not unique to it. Two concrete examples: first, when I was writing the transformation to create me/fare_per_pax, it wasn't super clear what needed to happen in the sky file and what needed to happen in the dataset.yml. I started out with both and had to deal with a bunch of errors which stemmed from (I think) me accidentally specifying things in both places or in the wrong place. In particular I wasn't sure what runs first. For example, does the ds in the transformation know the schema specified in dataset.yml or does qri check the output of transform against the specified schema?
+
+@dustmop pointed directly at the issue this feedback raised [while we were working on that rfc](https://github.com/qri-io/rfcs/pull/25#discussion_r225723031):
+
+> The feedback also points to a possible ambiguity between transform and dataset.yml, since both can do similar things. This is a distinct (but admittedly related) problem from the semantics of `qri save`. I understand it's hard to answer the original question without cleaning up our mental model, but I think it's also important to address the original question directly.
+
+Both are right. Words like "confusion" and "ambiguity" mean lack of clarity. Qri is emphasizing brevity too much, and needs to be more clear, even if it means adding a step or two. To do this properly, each added step should reinforce the mental model of _applying changes to a dataset_.
+
+There's also  a technical motivation for making this change as well What's worse, this "too many things" is costing us reproducibility. Because multiple things can act on a dataset to get form one snapshot to another, we can't accuractly reproduce histories by re-playing transformations. 
+
+## Proposal distinguish between "Manual" and "Scripted" transforms
+To me, the root of the problem here is we're being concise as the cost of clarity. it's clear that Qri is doing too many things in a single step, and allowing multiple things to affect the outcome of a `qri save`.  To solve this, we'll enforce some new rules and ratchet up some definitions. I want to start by adding formality to the the term "transform":
+
+### A "transform" is a forward transition from one dataset snapshot to another.
+* transforms must mutate one or more non-computed fields of a dataset
+* only one type of transform can be applied to any field per transform
+* transforms can use one or more types of mutations to determine the next snapshot
+
+A transform is the opposite of a history, moving forwards in snapshots instead of backwards. A history is _reproducible_ when you can start at the first snapshot and re-execute each mutation described in the next snapshot. By enforcing the mutually-exclusive mutations, each snapshot is a deterministic record of both state and _how to arrive at that state_ from the previous snapshot.
```

Comment:
I think it's important to add the assumption that external resources are unchanging. For example, we must assume that fetching some file over http will have the same result in the future. Of course, sometimes this assumption will fail, but that's outside of our control and there's nothing we can do about it. But as long as we either have datasets that don't rely on external resources and/or the assumption of unchanging externalities holds, we guarantee reproducibility.